starknet-replay/src/histogram/mod.rs

```diff
@@ -0,0 +1,281 @@
+    pub fn new(libfunc_stats: &ReplayStatistics) -> Result<Self, HistogramError> {
+        let max_frequency = libfunc_stats
+            .get_highest_frequency()
+            .ok_or(HistogramError::Empty)?;
+        let number_of_buckets = libfunc_stats.get_number_of_libfuncs();
+        let x_label_area = Self::calc_x_label_area(libfunc_stats)?;
+        let width = Self::calc_width(number_of_buckets)?;
+        let max_y_axis = Self::calc_max_y_axis(max_frequency)?;
+        let height = Self::calc_height(max_y_axis, x_label_area)?;
+
+        tracing::info!("Number of buckets {number_of_buckets}");
+        tracing::info!("Max y axis {max_y_axis}");
+
+        Ok(Config {
+            width,
+            height,
+            max_y_axis,
+            x_label_area,
+        })
+    }
+
+    /// Calculate the space required required to render the x axis labels.
+    ///
+    /// The space is returned in pixels.
+    ///
+    /// # Arguments
+    ///
+    /// - `libfunc_stats`: data to be plotted on the histogram.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if:
+    ///
+    /// - There is a math overflow when computing the number of pixels.
+    /// - There is a truncation when casting from `usize` to `u32`.
+    fn calc_x_label_area(libfunc_stats: &ReplayStatistics) -> Result<Pixel, HistogramError> {
+        let chars_longest_name: usize = libfunc_stats
+            .get_libfuncs()
+            .iter()
+            .max_by_key(|p| p.len())
+            .unwrap_or(&"")
+            .len();
+        let pixels_per_char: usize = 15;
+        let x_label_area_size: usize =
+            chars_longest_name
+                .checked_mul(pixels_per_char)
+                .ok_or(HistogramError::MathOverflow(
+                    "calc_x_label_area".to_string(),
+                ))?;
+        Ok(u32::try_from(x_label_area_size)?)
+    }
+
+    /// Calculate the maximum number shown on the y-axis of the histogram.
+    ///
+    /// The principle is to automatically resize the axis depending on the
+    /// number of times the most frequently called `libfunc` is called. This
+    /// number is rounded to the next hundreds.
+    ///
+    /// # Arguments
+    ///
+    /// - `max_frequency`: The highest frequency to be represented in the
+    ///   histogram.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`Err`] if there is an overflow in the calculation of the y-axis
+    /// extension.
+    fn calc_max_y_axis(max_frequency: usize) -> Result<usize, HistogramError> {
+        // `div` and `add` don't need to be checked because they will always return a
+        // number less than `max_frequency`, therefore fitting in the size of `usize`.
+        let max_y_axis = max_frequency.div(100).add(1).checked_mul(100);
+        max_y_axis.ok_or(HistogramError::MathOverflow("calc_max_y_axis".to_string()))
+    }
+
+    /// Calculate the width in pixels of the SVG image containing the histogram.
```

Comment:
I don't really understand what this function is doing. An SVG's resolution is entirely arbitrary.